shell.go

```diff
@@ -72,16 +73,25 @@ func NewShell(url string) *Shell {
 	return NewShellWithClient(url, c)
 }
 
-func NewShellWithClient(url string, c *gohttp.Client) *Shell {
-	if a, err := ma.NewMultiaddr(url); err == nil {
-		_, host, err := manet.DialArgs(a)
-		if err == nil {
-			url = host
+func NewShellWithClient(url string, client *gohttp.Client) *Shell {
+	if maddr, err := ma.NewMultiaddr(url); err == nil {
+		if network, host, err := manet.DialArgs(maddr); err == nil {
+			if network == "unix" {
+				url = network
+				if tpt, ok := client.Transport.(*gohttp.Transport); ok && tpt != nil && tpt.DialContext == nil {
+					tpt.DialContext = func(_ context.Context, _, _ string) (net.Conn, error) {
```

Comment:
We should copy the client & transport. We store the client by-value specifically because we modify it and don't want to modify the caller's copy.